### DISCUSSION TITLE
SSE events not listening

### DISCUSSION THREAD

**Discussion by gregcatalan** [cypress-io/cypress#29537] [Questions and Help]
## Current behavior:
We want to test a Live Tail that is supposed to receive messages in real time. As we cannot use multiple tabs/browsers, we send the message by cy.request(). This process works properly as we could verify where the messages are being stored. Nonetheless, the message does not appear in the Live Tail.

We have a SSE that is supposed to listen the calls. Manually, it receives in regular intervals something like this before the message is received:

```
event: tailBody
data: {}
```

After receiving the message, the SSE is supposed to receive responses like this and reflect it in a table:

```
event: tailBody
data: {"offsets":["XXXXXX,X,X"]}
```

On the other hand, in Cypress this SSE does not receive any responses and the table remains empty. We could open the same link in parallel to verify the process manually and it works properly.

The status of the service finishes as (failed)net::ERR_EMPTY_RESPONSE.

## Desired behavior:
The SSE should be listening for responses.

## Versions
Cypress v13.9.0
Chrome 124

Please let me know if I can provide further information.

**Comment by portyu9** [comment]
Hi @gregcatalan 

This looks like a Cypress/EventSource limitation rather than a problem with your `cy.request()`.

Since you confirmed that `cy.request()` successfully creates the backend message, the failing part is the browser's long-lived SSE connection. The fact that:

* the SSE receives no heartbeat events under Cypress,
* the same stream works outside Cypress, and
* Cypress reports `net::ERR_EMPTY_RESPONSE`

matches the long-standing Cypress EventSource/SSE issue (#2747).

I would not try to solve this with additional waits or `cy.intercept()`. `cy.intercept()` is not a reliable way to consume a streaming `EventSource` response because an SSE response intentionally remains open.

For the Cypress UI test, a practical approach is to replace `window.EventSource` before the application loads and emit the SSE events you need:

```js
cy.visit('/', {
  onBeforeLoad(win) {
    win.EventSource = FakeEventSource
  },
})
```

Then trigger a `tailBody` event and assert that the Live Tail table updates.

Keep a separate integration test for the real SSE endpoint if you need to verify the backend stream itself.

If the requirement is specifically a true end-to-end test where the real browser `EventSource` remains connected to the real backend, Cypress is currently not a reliable choice for that scenario; use a browser automation path that can preserve the native SSE connection.

Your use of `cy.request()` to create the backend message is fine—the evidence points to the receiving SSE connection, not the request that generates the message.

**Reply by portyu9** [reply]
@gregcatalan Please mark this answer as accepted if it helped, thank you.
